### DISCUSSION TITLE
Streamlit not working in codespaces

### DISCUSSION THREAD

**Discussion by batmanscode** [community/community#18038] [Codespaces]
Streamlit installed, port forwarded and successfully opens the Streamlit app - but it gets stuck on `Please wait...`

How to reproduce:
1. Create new codespaces
2. `pip install streamlit`
3. `streamlit hello`
4. Select open in browser on the port forwarding popup

Streamlit used to run on codespaces: https://discuss.streamlit.io/t/development-on-github-codespace/6830

**Comment by batmanscode** [comment]
Looks like it's a problem with CORS and/or XSRF in codespaces. Streamlit has these enabled by default and this can be turned off with `streamlit hello --server.enableCORS false --server.enableXsrfProtection false` at launch or via `config.toml`

Discussion here: https://discuss.streamlit.io/t/how-to-make-streamlit-run-on-codespaces/24526/5

UPDATE: it doesn't always work and always fails for me when I run it immediately after the code space is set up or with a command in devcontainer

**Reply by batmanscode** [reply]
UPDATE 2: I still can't get it to work

**Comment by sunchojack** [comment]
for me it was the issue with mozilla enhanced tracking protection, disabling helped
![image](https://github.com/user-attachments/assets/e43d1b00-c1e2-4fd1-8251-a519aa2fc279)

**Reply by batmanscode** [reply]
Can you try this: `streamlit run app.py --server.enableCORS false --server.enableXsrfProtection false`

Fixed it for me and is how I always use streamlit in codespaces now

**Comment by plp940** [comment]
tried all suggested commands but still i am getting bash: streamlit: command not found response, and downloaded all requirements too
